File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();

        for (int num : nums) {
            if (minHeap.size() < k || minHeap.peek() < num) {
                minHeap.offer(num);
                if (minHeap.size() > k) {
                    minHeap.poll();
                }
            }
        }

        return minHeap.peek();
    }
}

File: 215-kth-largest-element-in-an-array/kth-largest-element-in-an-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findKthLargest = function (nums, k) {
    const minHeap = new MinPriorityQueue();

    for (const num of nums) {
        if (minHeap.size() < k || minHeap.front() < num) minHeap.enqueue(num);
        if (minHeap.size() > k) minHeap.dequeue();
    }

    return minHeap.front();
};
